app/controllers/trainer_controller.rb

```diff
@@ -3,8 +3,19 @@ class TrainerController < ApplicationController
 
   def review_card
     @card = current_user.cards.find(params[:card_id])
-    if @card.check_translation(trainer_params[:user_translation])
-      flash[:notice] = 'Вы ввели верный перевод. Продолжайте.'
+
+    check_result = @card.check_translation(trainer_params[:user_translation])
+
+    if check_result[:state]
+      if check_result[:distance] == 0
+        flash[:notice] = 'Вы ввели верный перевод. Продолжайте.'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.